pilot/pkg/config/kube/crd/controller/controller.go

```diff
@@ -86,6 +86,25 @@ func init() {
 	monitoring.MustRegister(k8sEvents, k8sErrors, k8sTotalErrors)
 }
 
+// crdExistsWithRetry checks if the provided CRD exists
+// Any errors are retried
+func crdExistsWithRetry(client *Client, schema collection.Schema) bool {
+	delay := time.Second
+	maxDelay := time.Minute
```

Comment:
A minute seems high. istiod startup could block for multiple minutes given "its perfectly reasonable to not have the CRDs if you are not planning to use them". If the user isn't planning on using them, a longer wait time shouldn't matter, right? I think the retries is primarily to handle any races during start-up.

If the longer per-type timeout is retained, it could reduced calling crdExistsWithEntry concurrently for all types.

Reply:
Won't the long time conflict with readiness probes since this is called from the init path? 

Also, the `stopCh` should be plumbed through for faster early cancellation if the pod is killed early.

Reply:
This is not polling for the CRD, if it gets ErrNotFound it exist immediately. In CRDExists function it handles this. so this should just be for transient errors to connect to api server

Reply:
Ah, yes - nevermind then.